### PR TITLE
Expand contents of Integrations Developer Guide

### PR DESCRIPTION
**Pull request by kilfoyle**
This updates the Integrations Developer Guide with some of the docs contents in the [integrations repo](https://github.com/elastic/integrations/tree/main/docs):

screen

**Previews:**
 - [Integrations Guidelines](https://observability-docs_bk_3836.docs-preview.app.elstc.co/guide/en/integrations-developer/current/integrations-guidelines.html) section
 - [Working with new indexing features](https://observability-docs_bk_3836.docs-preview.app.elstc.co/guide/en/integrations-developer/current/integrations-tsds-synthetic-source.html) section
 - [Definitions](https://observability-docs_bk_3836.docs-preview.app.elstc.co/guide/en/integrations-developer/current/integration-definitions.html) page
 - [New sections](https://observability-docs_bk_3836.docs-preview.app.elstc.co/guide/en/integrations-developer/current/testing-and-validation.html#_review_test_coverage) in Testing and validataion page

Closes: https://github.com/elastic/ingest-docs/issues/1032

### CODE REVIEW

File: docs/en/integrations/integrations-guidelines.asciidoc [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+* Provide a comprehensive list of collected fields and the data and metric types for each. The reader can reference this information while evaluating the integration, interpreting collected data, or troubleshooting issues.
+* Set the reader up for a successful installation and setup by connecting them with any other resources they'll need.
+* Each integration doc should contain several sections, and you should use consistent headings to make it easier for a single user to evaluate and use multiple integrations.
+
+* <<idg-docs-guidelines-overview>>
+* <<idg-docs-guidelines-datastreams>>
+* <<idg-docs-guidelines-requirements>>
+* <<idg-docs-guidelines-setup>>
+* <<idg-docs-guidelines-troubleshooting>>
+* <<idg-docs-guidelines-reference>>
```
Comment (bmorelli25) [inline docs/en/integrations/integrations-guidelines.asciidoc:359]: This all renders as one ordered list. I think you mean for the second part to break?

File: docs/en/integrations/integrations-guidelines.asciidoc [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+
+Some considerations when these documentation files are written at `_dev/build/docs/*.md`:
+
+* These files follow the Markdown syntax and leverage the use of link:https://github.com/elastic/elastic-package/blob/main/docs/howto/add_package_readme.md[documentation templates].
+* There are some available functions or placeholders (`fields`, `event`, `url`) that can be used to help writing these docs. For more detail, refer to link:https://github.com/elastic/elastic-package/blob/main/docs/howto/add_package_readme.md#placeholders[placeholders].
+- Regarding the `url` placeholder, this placeholder should be used to add links to the link:https://www.elastic.co/guide/index.html[Elastic documentation guides] in your documentation:
+    - File containing all the links defined is in the root of the directory: [`links_table.yml`](../links_table.yml)
+    - If needed, more links to Elastic documentation guides can be added into that file.
+    - Example usage:
+        - In the documentation files (`_dev/build/docs/*.md`) `{{ url "getting-started-observability" "Elastic guide" }}` generates a link to the Observability Getting Started guide.
```
Comment (bmorelli25) [inline docs/en/integrations/integrations-guidelines.asciidoc:369]: This doesn't render correctly

Comment (bmorelli25) [reply docs/en/integrations/integrations-guidelines.asciidoc:369]:
```suggestion
* Regarding the `url` placeholder, this placeholder should be used to add links to the link:https://www.elastic.co/guide/index.html[Elastic documentation guides] in your documentation:
** File containing all the links defined is in the root of the directory: [`links_table.yml`](../links_table.yml)
** If needed, more links to Elastic documentation guides can be added into that file.
** Example usage:
*** In the documentation files (`_dev/build/docs/*.md`) `{{ url "getting-started-observability" "Elastic guide" }}` generates a link to the Observability Getting Started guide.
```

**Review by bmorelli25 — COMMENTED** [review]
Looking good so far! A couple small things I noticed when clicking through

**Comment by agithomas** [comment]
The [TSDB guidelines document](https://github.com/elastic/integrations/blob/main/docs/developer_tsdb_migration_guidelines.md) was prepared originally to assist the Elastic (internal) developers to migrate from Non-TSDS -> TSDS based datastreams.


Considering this, there are a few suggestions

1. Please use the name TSDS instead of TSDB. [TSDS](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds.html) is the preferred terminology by the Elasticsearch team.
2. Please have a reference to [TSDS documentation ](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds.html) to help users aware of the features of TSDS
3. The phrase `Meta Issue with all migrated packages` can be excluded as the migration task is already completed.
4. The heading2 `Steps for migrating an existing package` may be changed to `Steps for enabling TSDS for a metrics dataset.
5. The related resource (TSDB [test](https://github.com/elastic/TSDB-migration-test-kit) migration kit), may be referred to in the Troubleshooting section as a utility to debug a situation when there are document drops post the TSDS enablement.

**Comment by kilfoyle** [comment]
Thanks a lot for the excellent feedback @agithomas! I'll get those changes in asap.

**Comment by kilfoyle** [comment]
@agithomas Great feedback! I've addressed all of your comments. You can find the latest version of the ~~TSDB~~TSDS steps [here](https://observability-docs_bk_3836.docs-preview.app.elstc.co/guide/en/integrations-developer/current/developer-tsds-guidelines.html).

**Comment by agithomas** [comment]
@kilfoyle , the changes look good to me!

++ @lalit-satapathy

**Comment by bmorelli25** [comment]
Thank you Agi and David! I'm not sure if Lalit has had a chance to look at this yet. I'll merge in 24 hours 🎉 .

**Comment by alaudazzi** [comment]
@kilfoyle Thank you for compiling some of the pieces of what seems to be a big puzzle.
I need more time to review this content reorg, but I don't want to block the merge of this PR. 
I'll get back to this portion of the Integrations docs soon and provide feedback where needed.

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**